### PR TITLE
[106] Encrypt JSON exports

### PR DESCRIPTION
**Pull request by maelchiotti**
- [x] Add encryption utils
- [x] Add JSON exports encryption
- [x] Add JSON imports decryption

### CODE REVIEW

**Comment by maelchiotti** [comment]
Hi @HarriBuh and @super-lurker!

I've worked on allowing the JSON exports to be encrypted, both the manual ones and the new (to be released) automatic ones. You can specify a 32 characters long, strong passphrase that will be used to perform AES encryption on the title and the content of the notes. Of course, the same passphrase is needed when importing back the export.

I you wish to test this out, I'd love to get your feedback, so feel free to grab the .apk in the artifacts of the deployment just below. And to be safe, backup your notes first of course ☺️

**Comment by super-lurker** [comment]
looks good to me 👍

File: lib/common/widgets/notes_list.dart [hunk 1]
```diff
@@ -16,21 +15,25 @@ import 'package:localmaterialnotes/utils/preferences/layout.dart';
...
   @override
   Widget build(BuildContext context) {
-    final isNotes = RouterRoute.currentRoute == RouterRoute.notes;
+    final isNotes = widget.route == RouterRoute.notes;
     final provider = isNotes ? notesProvider : binProvider;
 
     return ref.watch(provider).when(
       data: (notes) {
-        if (notes.isEmpty) {
-          return isNotes ? EmptyPlaceholder.notes() : EmptyPlaceholder.bin();
```
Comment (maelchiotti) [inline lib/common/widgets/notes_list.dart:32]: Add empty placeholders back

File: lib/pages/settings/settings_actions.dart [hunk 2]
```diff
@@ -141,16 +143,65 @@ class SettingsActions {
...
+      final frequency = autoExportSettings.$1;
+      PreferencesUtils().set<double>(PreferenceKey.autoExportFrequency.name, frequency);
+
+      // If the auto export was disabled, just remove the encryption and passphrase settings
+      if (frequency == 0.0) {
+        await PreferencesUtils().remove(PreferenceKey.autoExportEncryption);
+        await PreferencesUtils().deleteSecure(PreferenceKey.autoExportPassphrase);
+
+        return;
+      }
```
Comment (maelchiotti) [inline lib/pages/settings/settings_actions.dart:164]: Also remove the last export date

**Pull request merged**